tests/integrationv2/test_record_padding.py

```diff
@@ -0,0 +1,171 @@
+import copy
+import pytest
+import re
+
+from configuration import available_ports, TLS13_CIPHERS, ALL_TEST_CURVES, MINIMAL_TEST_CERTS
+from common import ProviderOptions, Protocols, data_bytes
+from fixtures import managed_process  # lgtm [py/unused-import]
+from providers import Provider, S2N, OpenSSL
+from utils import invalid_test_parameters, get_parameter_name, get_expected_s2n_version, to_bytes
+
+PADDING_SIZE_SMALL = 250
+PADDING_SIZE_MEDIUM = 1000
+PADDING_SIZE_MAX = 1 << 14
+
+PADDING_SIZES = [
+    PADDING_SIZE_SMALL,
+    PADDING_SIZE_MEDIUM,
+    PADDING_SIZE_MAX
+]
+
+# arbitrarily large payload size
+PAYLOAD_SIZE = 1024
+
+# regex that matches server/client 'write to' with the TLS header
+OPENSSL_WRITE_TO_REGEX = r"write to [\w >\(\)\[\]=]*\\n0000 - 17 03 03 [0-9A-Fa-f]{2} [0-9A-Fa-f]{2}"
```

Comment:
0x17 == 23 == APPLICATION_DATA. While most of the handshake records are disguised as application data and will match this header, the first few won't. The ChangeCipherSpec and any alerts also won't.

Reply:
How rigorous do we need this check to be? This is meant to be a sanity check that openssl is padding as expected; checking all the sent application data and their sizes should be sufficient.

Reply:
I don't think we necessarily NEED it to catch all records, but does it cost us anything to relax the check? Does the 17 simplify something?

Reply:
Yes, checking just application data makes it easier. This is because for records of type 0x17, we know the record size is `n * padding_size + 16` where `n` is some integer. 16 is known because we're using tls 13 ciphersuites with 16 byte tags.

For a change cipher suite, the record size different; this is because this record also has the MAC inside (struct reproduced from rfc below). By checking just records of type `0x17` we avoid branching checks and we can still be confident openssl is padding as expected

```
struct {
    opaque IV[SecurityParameters.record_iv_length];
    block-ciphered struct {
        opaque content[TLSCompressed.length];
        opaque MAC[SecurityParameters.mac_length]; // <-- server reads this fine!
        uint8 padding[GenericBlockCipher.padding_length];
        uint8 padding_length;
    };
} GenericBlockCipher;
```

Reply:
Records up to and including the ChangeCipherSpec aren't encrypted. They're plaintext. But that would explain your problem; if they're not encrypted, no tag.

Reply:
But that's fair. Checking just ApplicationData records should give us confidence that Openssl is setup properly.